samples/nrf_desktop/src/modules/usb_state.c

```diff
@@ -63,73 +101,76 @@ static int set_report(struct usb_setup_packet *setup, s32_t *len, u8_t **data)
 {
 	if (IS_ENABLED(CONFIG_DESKTOP_CONFIG_CHANNEL_ENABLE)) {
 		size_t length = *len;
-		const u8_t *buffer = *data;
+		u8_t *buffer = *data;
 
-		u8_t  report_id;
-		u16_t recipient;
-		u8_t  event_id;
-		u8_t  event_data_len;
-
-		const size_t min_size = sizeof(report_id) + sizeof(recipient) +
-					sizeof(event_id) + sizeof(event_data_len);
-		const size_t max_size = sizeof(report_id) + REPORT_SIZE_USER_CONFIG;
+		if (atomic_get(&forward_status) == FORWARD_STATUS_PENDING) {
+			LOG_WRN("Busy forwarding");
+			return -EBUSY;
+		}
 
-		static_assert(min_size < max_size, "");
+		/* Feature report set */
+		int pos = config_channel_report_parse(buffer, length, &frame, true);
+		if (pos < 0) {
+			LOG_WRN("Could not parse report");
+			return pos;
+		}
 
-		if ((length < min_size) || (length > max_size)) {
-			LOG_WRN("Unsupported report length %zu", length);
+		if (frame.report_id != REPORT_ID_USER_CONFIG) {
+			LOG_WRN("Unsupported report ID %" PRIu8, frame.report_id);
 			return -ENOTSUP;
 		}
 
-		if (atomic_get(&forward_status) == FORWARD_STATUS_PENDING) {
-			LOG_WRN("Busy forwarding");
-			return -EBUSY;
-		}
+		if (frame.recipient == CONFIG_USB_DEVICE_PID) {
+			if (atomic_get(&fetch_done)) {
+				LOG_WRN("Pending fetched data discarded");
```

Comment:
This is ok but the other device does not know about the new transaction. It may still work to complete the old request and when it is done the data will be returned as an answer to the new request which is clearly wrong.
A way to fix it would be to add a transaction tag in the request.

Reply:
only one transaction at a time can be now processed. Also both firmware and host check if the correct event_id and recipient is provided back to host